### PR TITLE
✨ Added twitter login

### PR DESCRIPTION
**Pull request by soulcramer**
Twitter provide an OAuth2 authentification only for applications which is not what we want (we need user informations) which is why we need to use the OAuth1 authentification.

### CODE REVIEW

**Comment by soulcramer** [comment]
This should also close #161

File: auth/twitter.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+	// If both Twitter ID and email can't be found in our DB, register a new user.
+	// Otherwise, log in the user with the given Twitter ID or email.
+	app.Get("/auth/twitter/callback", func(ctx *aero.Context) string {
+		if !ctx.HasSession() {
+			return ctx.Error(http.StatusUnauthorized, "Twitter login failed", errors.New("Session does not exist"))
+		}
+
+		session := ctx.Session()
+
+		// get back the request token to get the access token
```
Comment (akyoto) [inline auth/twitter.go:66]: Comment starts with a lowercase letter. This is inconsistent.

File: auth/twitter.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+			session.Set("userId", user.ID)
+			return ctx.Redirect("/")
+		}
+
+		// Register new user
+		user = arn.NewUser()
+		user.Nick = "tw" + twUser.ID
+		user.Email = twUser.Email
+		// The full name is in Name and separated like other services
```
Comment (akyoto) [inline auth/twitter.go:149]: https://github.com/animenotifier/notify.moe/blob/go/docs/code-style.md#empty-line-between-commented-blocks

File: auth/twitter.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+		// Get the request token from twitter
+		tokenCred, _, err := config.RequestToken(nil, tempCred, ctx.Query("oauth_verifier"))
+
+		if err != nil {
+			return ctx.Error(http.StatusBadRequest, "Could not obtain OAuth access token", err)
+		}
+
+		// Fetch user data from Twitter
+		resp, err := config.Get(nil, tokenCred, "https://api.twitter.com/1.1/account/verify_credentials.json?include_email=true1skip_status=true", nil)
```
Comment (akyoto) [inline auth/twitter.go:81]: Is this URL actually correct or is the "1" a typing mistake?

**Pull request merged**